pubtools/_pulp/tasks/publish.py

```diff
@@ -0,0 +1,190 @@
+import logging
+import sys
+import re
+from datetime import datetime
+from functools import partial
+from more_executors.futures import f_map, f_sequence
+
+from pubtools.pulplib import Criteria, Matcher, PublishOptions
+
+from pubtools._pulp.task import PulpTask, CDNCached
+from pubtools._pulp.services import PulpClientService, UdCacheClientService
+
+
+step = PulpTask.step
+
+LOG = logging.getLogger("pubtools.pulp")
+
+
+def publish_date(str_date):
+    # validates publish-before date
+    return datetime.strptime(str_date, "%Y-%m-%d")
+
+
+class Publish(PulpClientService, UdCacheClientService, PulpTask, CDNCached):
+    """Publishes the pulp repositories to the endpoints defined by the distributors
+
+    This command will publish the pulp repositories provided in the request or
+    fetched using the filters(url-regex or published-before) or an intersection
+    of input repos and filters.
+    """
+
+    def add_args(self):
+        super(Publish, self).add_args()
+
+        self.parser.add_argument(
+            "--repo-ids", help="list of repos to be published", nargs="+"
+        )
+        self.parser.add_argument(
+            "--clean",
+            help="attempt to remove content not in the repo",
+            action="store_true",
+        )
+        self.parser.add_argument(
+            "--force", help="execute a full repo publish", action="store_true"
+        )
+        self.parser.add_argument(
+            "--published-before",
+            help="publish the repos last published before given date e.g. 2019-08-21",
+            type=publish_date,
+            default=None,
+        )
+        self.parser.add_argument(
+            "--repo-url-regex",
+            help="publish repos whose repo url match",
+            default=None,
+            type=re.compile,
+        )
+
+    def run(self):
+        to_await = []
+        LOG.info("run publish")
```

Comment:
This doesn't look like a meaningful message for end-users.  Could it please be removed, changed to a DEBUG or changed to something useful?